libs/smart-ngrx/src/selector/array-proxy.class.ts

```diff
@@ -21,7 +21,7 @@ import { isArrayProxy } from './is-array-proxy.function';
  * @see `createSmartSelector`
  */
 export class ArrayProxy<
-    P extends object = object,
+    P extends SmartNgRXRowBase = SmartNgRXRowBase,
     C extends SmartNgRXRowBase = SmartNgRXRowBase,
   >
```

Comment:
In this case, it is intentional because we are using the Proxy class to intercept and it is the Proxy that we want the constructor to return.

Is there a better way to achieve this goal?

Reply:
Not really. I had done that before and it didn't completely serve my needs.

Reply:
comment added